crc-idle.py

```diff
@@ -65,14 +65,17 @@ def _idle_cpu_resources(self, cluster, partition):
         return_dict = dict()
         for node_info in slurm_data:
             node_name, resource_data = node_info.split(',')
+
             # Return values include: allocated, idle, other, total
             _, idle, _, _ = [int(x) for x in resource_data.split('/')]
-            return_dict[idle] += 1
+            
+            return_dict[idle] = return_dict.setdefault(idle,0) + 1
```

Comment:
I think the issue I was having here was that `return_dict[idle] += 1` was erroring when a node was found to have a number of idle cores that was not already in the dictionary. I think there was a `setdefault` line that had been removed, but it should be okay to use as it sets the value to the default only in that case that was causing the error and increments the count properly otherwise.

Reply:
Yes, `setdefault` only sets the value once. I meant to indicate that

```python
return_dict.setdefault(idle,0)
return_dict[idle] += 1
```

is not functionally equivalent to 

```python
return_dict[idle] = return_dict.setdefault(idle,0) + 1 
```

Since `return_dict` returns the value and not the instance of the dictionary, the later example does not update the dictionary contents. For example:

```python
>>> example = dict()
>>> example.setdefault('a', 0) + 1
1
>>> example
{'a': 0}
```

Reply:
The first two code blocks do achieve the same effect: 
If a key does not exist yet, set it's value to 0 and increment it, otherwise increment it's current value. Your example is doing `example.setdefault('a',0) + 1`, not `example['a'] = setdefault('a',0) + 1`. 

Both the assignment and the `setdefaults` are necessary, the former covers the increment (+1 node with `idle` number of cores/gpus found), and the latter prevents a KeyError if `idle` is a number of cores/gpus that we haven't seen already.